src/jit/codegen.h

```diff
@@ -1211,11 +1216,21 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
     void genReturn(GenTree* treeNode);
 
+    // clang-format off
+    void genStackPointerConstantAdjustment(ssize_t spDelta
+                                           X86_ARG(bool hideSpChangeFromEmitter)
+                                           X86_ARG(regNumber regTmp));
+    void genStackPointerConstantAdjustmentWithProbe(ssize_t spDelta
+                                                    ARM_ARG(regNumber regTmp)
+                                                    X86_ARG(bool hideSpChangeFromEmitter)
+                                                    X86_ARG(regNumber regTmp));
```

Comment:
Ugg...  optional arguments
At least reorder the optional args so that `hideSpChangeFromEmitter `always comes last. 
Also it seems we could just normal arguments and pass `REG_ZR `on ARM64 and `REG_NA `on X64.
You could eliminate `hideSpChangeFromEmitter`, by making it the default when regTmp is specified.
I''m pretty sure the optimizer would eliminate any extra overhead from these arguments

Reply:
I guess I don't share your distaste for optional arguments, if the alternative is either `#ifdef` or arguments that are not used on a platform (e.g., arm64, x64 here).

> You could eliminate hideSpChangeFromEmitter, by making it the default when regTmp is specified.

I could. The semantic seems a little circuitous: "why would a register imply hiding the SP change?" would be my thinking. Also, it's a hack: we should just fix the emitter to hide the change directly, without requiring a register.

Reply:
There is no real cost for:

" arguments that are not used on a platform (e.g., arm64, x64 here)."

as the optimizer can inline and eliminate them.

Reply:
> as the optimizer can inline and eliminate them.

My concern isn't the runtime overhead of extra arguments, it's the programmer's mental overhead of seeing things that aren't used on a particular platform. If I see an argument, I expect it to be needed and used. If it's not, that's annoying and possibly confusing. I think the existence of unnecessary arguments is worse than platform-specific function signatures.

Reply:
I still think that we can get rid of these (horrible) optional args and just always pass regTmp.

And pass REG_NA or REG_ZR when they are not really needed.  IMO that isn't confusing to the  programmer's mental model.  We already do that all over the pass, rather than overload a method with different number of args.